tests/validation/tests/v3_api/test_configmaps.py

```diff
@@ -0,0 +1,489 @@
+            print(" Command to display configmap value from container is: ")
+            print(command)
+            result = kubectl_pod_exec(pod_list[0], command)
+            assert result.decode("utf-8") == (list(keyvaluepair.values())[i])
+        elif workloadwitconfigmapasenvvar:
+            command = 'env'
+            result = kubectl_pod_exec(pod_list[0], command)
+            print(list(keyvaluepair.values())[i])
+            if list(keyvaluepair.values())[i] in result.decode("utf-8"):
+                assert True
+
+
+def delete_configmap(client, configmap, ns, keyvaluepair):
+
+    key = list(keyvaluepair.keys())[0]
+    print("Delete Configmap")
+    client.delete(configmap)
+
+    # Sleep to allow for the configmap to be deleted
+    time.sleep(5)
+    print("Config Map list after deleting config map")
+    configmapdict = client.list_configMap()
+    print(configmapdict)
+    print(configmapdict.get('data'))
+    if len(configmapdict.get('data')) > 0:
+        testdata = configmapdict.get('data')
+        print("TESTDATA")
+        print(testdata[0]['data'])
+        if key in testdata[0]['data']:
+            assert False
+        else:
+            assert True
+    if len(configmapdict.get('data')) == 0:
+        assert True
+
+    # Verify configmap is deleted by "kubectl get configmap" command
+    command = " get configmap " + configmap['name'] + " --namespace=" + ns.name
+    print("Command to obtain the configmap")
+    print(command)
+    result = execute_kubectl_cmd(command, json_out=False, stderr=True)
+    print(result)
+
+    print("Verify that the configmap does not exist "
+          "and the error code returned is non zero ")
+    if result != 0:
+        assert True
+
+
+def create_and_validate_workload_with_configmap_as_volume(p_client, configmap,
+                                                          ns, keyvaluepair):
+    workload_name = random_test_name("test")
+    # Create Workload with configmap as volume
+    mountpath = "/test"
+    volumeMounts = [{"readOnly": False, "type": "volumeMount",
+                     "mountPath": mountpath, "name": "vol1"}]
+    con = [{"name": "test1",
+            "image": TEST_IMAGE,
+            "volumeMounts": volumeMounts}]
+
+    configmapname = configmap['name']
+
+    volumes = [{"type": "volume", "name": "vol1",
+                "configMap": {"type": "configMapVolumeSource",
+                              "defaultMode": 256,
+                              "name": configmapname,
+                              "optional": False}}]
+
+    workload = p_client.create_workload(name=workload_name,
+                                        containers=con,
+                                        namespaceId=ns.id, volumes=volumes)
+    validate_workload_with_configmap(p_client, workload, "deployment",
+                                     ns.name, keyvaluepair,
+                                     workloadwithconfigmapasvolume=True)
+
+    return workload
```

Comment:
should we delete the workload instead of returning it?

Reply:
Done